pkg/api/handlers/libpod/artifacts.go

```diff
@@ -0,0 +1,335 @@
+	"github.com/containers/podman/v5/pkg/auth"
+	"github.com/containers/podman/v5/pkg/domain/entities"
+	"github.com/containers/podman/v5/pkg/domain/infra/abi"
+	domain_utils "github.com/containers/podman/v5/pkg/domain/utils"
+	libartifact_types "github.com/containers/podman/v5/pkg/libartifact/types"
+	"github.com/docker/distribution/registry/api/errcode"
+	"github.com/gorilla/schema"
+	"github.com/sirupsen/logrus"
+)
+
+func InspectArtifact(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+
+	name := utils.GetName(r)
+
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+
+	report, err := imageEngine.ArtifactInspect(r.Context(), name, entities.ArtifactInspectOptions{})
+	if err != nil {
+		if errors.Is(err, libartifact_types.ErrArtifactNotExist) {
+			utils.ArtifactNotFound(w, name, err)
+			return
+		} else {
+			utils.InternalServerError(w, err)
+			return
+		}
+	}
+
+	utils.WriteResponse(w, http.StatusOK, report)
+}
+
+func ListArtifact(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+
+	artifacts, err := imageEngine.ArtifactList(r.Context(), entities.ArtifactListOptions{})
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+
+	utils.WriteResponse(w, http.StatusOK, artifacts)
+}
+
+func PullArtifact(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+
+	query := struct {
+		Name       string             `schema:"name"`
+		Retry      uint               `schema:"retry"`
+		RetryDelay string             `schema:"retryDelay"`
+		TLSVerify  types.OptionalBool `schema:"tlsVerify"`
+	}{}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusBadRequest, fmt.Errorf("failed to parse parameters for %s: %w", r.URL.String(), err))
+		return
+	}
+
+	if query.Name == "" {
+		utils.Error(w, http.StatusBadRequest, errors.New("name parameter is required"))
+		return
+	}
+
+	artifactsPullOptions := entities.ArtifactPullOptions{}
+
+	// If TLS verification is explicitly specified (True or False) in the query,
+	// set the InsecureSkipTLSVerify option accordingly.
+	// If TLSVerify was not set in the query, OptionalBoolUndefined is used and
+	// handled later based off the target registry configuration.
+	switch query.TLSVerify {
+	case types.OptionalBoolTrue:
+		artifactsPullOptions.InsecureSkipTLSVerify = types.NewOptionalBool(false)
+	case types.OptionalBoolFalse:
+		artifactsPullOptions.InsecureSkipTLSVerify = types.NewOptionalBool(true)
+	default: // Should never happen, require TLS, to be conservative
+		logrus.Errorf("Unexpected handling occurred for TLSVerify, falling back to requiring TLS")
+		artifactsPullOptions.InsecureSkipTLSVerify = types.NewOptionalBool(false)
+	}
```

Comment:
This seems wrong and adds exactly the issue back I was mentioned before. The undefined case is not handled which means you hard code undefined to be false.
You would need an empty undefined case for this to make sense.

Then (a whole different topic) server side logs are basically never caught anywhere in CI tests which we won't notice this problem either. I think logically if we this cannot happen a panic() might be more reasonable to signal a invariant that should never happen.